tests/notification/run.py

```diff
@@ -1,174 +1,82 @@
-  except AttributeError as m:
-    attribute_err = m
-  except ValueError as m:
-    value_err = m
-
-  assert attribute_err is None, "Expected attribute for ch3 did not exist" \
-          + " (arg-one): %s" % attribute_err
-  assert value_err is None, "Expected value could not be set for ch3" \
-          + " (arg-one): %s" % value_err
-
-  attribute_err = None
-  value_err = None
-  try:
-    ch3.arguments.arg_two = "test string"
-  except AttributeError as m:
-    attribute_err = m
-  except ValueError as m:
-    value_err = m
-
-  assert attribute_err is None, "Expected attribute for ch3 did not exist" \
-          + " (arg-two): %s" % attribute_err
-  assert value_err is None, "Expected value could not be set for ch3" \
-          + " (arg-two): %s" % value_err
-
-  attribute_err = None
-  value_err = None
-  try:
-    ch4.arguments_one.arg_one = "test string"
-  except AttributeError as m:
-    attribute_err = m
-  except ValueError as m:
-    value_err = m
-
-  assert attribute_err is None, "Expected attribute for ch4 did not exist" \
-          + " (arg-one): %s" % attribute_err
-  assert value_err is None, "Expected value could not be set for ch4" \
-          + " (arg-one): %s" % value_err
-
-  attribute_err = None
-  value_err = None
-  try:
-    ch4.arguments_two.arg_two = "test string"
-  except AttributeError as m:
-    attribute_err = m
-  except ValueError as m:
-    value_err = m
-
-  assert attribute_err is None, "Expected attribute for ch4 did not exist" \
-          + " (arg-two): %s" % attribute_err
-  assert value_err is None, "Expected value could not be set for ch4" \
-          + " (arg-two): %s" % value_err
-
-  from bindings import notification
-
-  r = notification(path_helper=ph)
-  r.test.reference_target.append('five')
-
-  set_v = True
-  try:
-    ch5.argument = 'five'
-  except ValueError:
-    set_v = False
-
-  assert set_v is True, "Could not set value of a leafref in a notification to a" + \
-          "known good value: %s != True (ch.argument -> five)" % (set_v)
-
-  set_v = True
-  try:
-    ch5.argument = 'fish'
-  except ValueError:
-    set_v = False
-
-  assert set_v is False, "Set value of a leafref in a notification to a" + \
-          "known bad value: %s != False (ch.argument -> fish)" % (set_v)
-
-  if not k:
-    os.system("/bin/rm -rf %s/bindings" % this_dir)
+import unittest
+
+from pyangbind.lib.xpathhelper import YANGPathHelper
+from tests.base import PyangBindTestCase
+
+
+class NotificationTests(PyangBindTestCase):
+  yang_files = ['notification.yang']
+  split_class_dir = True
+  pyang_flags = ['--use-xpathhelper', '--build-notifications']
+
+  def setUp(self):
+    self.path_helper = YANGPathHelper()
+
+  # Note that these tests need to import from the filesystem rather than using
+  #  the bound module, so that they can directly reference the sub-modules
+  def test_set_leaf_inside_notification(self):
+    from .bindings.notification_notification.alert_one import alert_one
```

Comment:
Looks like this consistently causes a test failure because the test file isn't a module.